### ISSUE TITLE
gftools packager not yet working to make a PR – what am I missing?

### ISSUE THREAD

**Issue by arrowtype** [googlefonts/gftools#253]
I am trying to use the `packager` to make a PR of the Fraunces family, currently being mastered at https://github.com/arrowtype/Fraunces/tree/d98e84da8eaa4f59e99b4ed99e80246444e9819a.

[Here’s my current `upstream.yaml` file.](https://github.com/arrowtype/Fraunces/blob/mastering/mastering/googlefonts/upstream.yaml)

I’ve tried it both with the origin URL and with a `local` URL. I have tried with the origin repo URL plus the `--push` option a few times in slightly different ways, but have not yet successfully managed to deploy a PR.

### Questions

1. Do I need push access to google/fonts on my `@arrowtype` account for this to open PRs? If so, can I please get that?
2. Also, is it normal for the packager to clone the entire font project repo on each run I don’t use the `local` hack? Fraunces includes a lot of UFOs, so this takes up a lot of time on each run (maybe 3 minutes or more of just downloading data I already have). It seems from the docs that if I use the `local` hack, it won’t make a PR: `CAUTION: Google Fonts won't accept families where the METADATA.pb source.repository_url and source.commit hash are not publicly available.`

> `Receiving objects:   1% (371/22420), 66.01 MiB | 275.00 KiB/s`

**Comment by arrowtype** [comment]
Oh, it finally worked on my latest try, and it does look like I’m lacking access permissions to make the PR:

```
Making a Pull Request …
git push:
  url is git@github.com:google/fonts.git
  local branch name is gftools_packager_ofl_fraunces
  remote branch name is gftools_packager_ofl_fraunces

ERROR: Permission to google/fonts.git denied to arrowtype.
fatal: Could not read from remote repository.

Please make sure you have the correct access rights
and the repository exists.
Traceback (most recent call last):
  File "/Users/stephennixon/type-repos/google-font-repos/Fraunces/venv/bin/gftools-packager.py", line 144, in <module>
    packager.make_package(**args.__dict__)
  File "/Users/stephennixon/type-repos/google-font-repos/Fraunces/venv/lib/python3.7/site-packages/gftools/packager.py", line 1722, in make_package
    _dispatch_git(target, target_branch, pr_upstream, push_upstream)
  File "/Users/stephennixon/type-repos/google-font-repos/Fraunces/venv/lib/python3.7/site-packages/gftools/packager.py", line 1478, in _dispatch_git
    pr_title, pr_message_body)
  File "/Users/stephennixon/type-repos/google-font-repos/Fraunces/venv/lib/python3.7/site-packages/gftools/packager.py", line 1193, in _make_pr
    _push(repo, url, local_branch_name, remote_branch_name, force=True)
  File "/Users/stephennixon/type-repos/google-font-repos/Fraunces/venv/lib/python3.7/site-packages/gftools/packager.py", line 1119, in _push
    stdout=subprocess.PIPE)
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/subprocess.py", line 487, in run
    output=stdout, stderr=stderr)
subprocess.CalledProcessError: Command '['git', 'push', 'git@github.com:google/fonts.git', '+refs/heads/gftools_packager_ofl_fraunces:refs/heads/gftools_packager_ofl_fraunces']' returned non-zero exit status 128.
```

**Comment by arrowtype** [comment]
I also tried via my @thundernixon account, but it seems like access has probably been revoked there (which is fine, as I am trying to wind down that account)

```
Making a Pull Request …
git push:
  url is git@github.com:google/fonts.git
  local branch name is gftools_packager_ofl_fraunces
  remote branch name is gftools_packager_ofl_fraunces

ERROR: Permission to google/fonts.git denied to thundernixon.
fatal: Could not read from remote repository.

Please make sure you have the correct access rights
and the repository exists.
Traceback (most recent call last):
  File "/Users/stephennixon/type-repos/google-font-repos/Fraunces/venv/bin/gftools-packager.py", line 144, in <module>
    packager.make_package(**args.__dict__)
  File "/Users/stephennixon/type-repos/google-font-repos/Fraunces/venv/lib/python3.7/site-packages/gftools/packager.py", line 1722, in make_package
    _dispatch_git(target, target_branch, pr_upstream, push_upstream)
  File "/Users/stephennixon/type-repos/google-font-repos/Fraunces/venv/lib/python3.7/site-packages/gftools/packager.py", line 1478, in _dispatch_git
    pr_title, pr_message_body)
  File "/Users/stephennixon/type-repos/google-font-repos/Fraunces/venv/lib/python3.7/site-packages/gftools/packager.py", line 1193, in _make_pr
    _push(repo, url, local_branch_name, remote_branch_name, force=True)
  File "/Users/stephennixon/type-repos/google-font-repos/Fraunces/venv/lib/python3.7/site-packages/gftools/packager.py", line 1119, in _push
    stdout=subprocess.PIPE)
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/subprocess.py", line 487, in run
    output=stdout, stderr=stderr)
subprocess.CalledProcessError: Command '['git', 'push', 'git@github.com:google/fonts.git', '+refs/heads/gftools_packager_ofl_fraunces:refs/heads/gftools_packager_ofl_fraunces']' returned non-zero exit status 128.
```

**Comment by vv-monsalve** [comment]
> 2\. Also, is it normal for the packager to clone the entire font project repo on each run I don’t use the `local` hack? Fraunces includes a lot of UFOs, so this takes up a lot of time on each run (maybe 3 minutes or more of just downloading data I already have). It seems from the docs that if I use the `local` hack, it won’t make a PR: `CAUTION: Google Fonts won't accept families where the METADATA.pb source.repository_url and source.commit hash are not publicly available.`

Which command are you using?

**Comment by arrowtype** [comment]
I used:

```
gftools packager mastering/googlefonts/upstream.yaml ~/type-repos/google-font-repos/fonts -fpy
```

**Comment by vv-monsalve** [comment]
_First, a disclaimer: I'm also new at packager, starting to use it_

The first thing I notice is the .yaml doesn't have the family name, which from what I understood is needed as the tool would, by default, create/use a branch named out of the family name.

After that, and particularly to start using the tool I wouldn't use the flag `-f` and `-y`. In this way, the tool is going to inform you what is needed as it goes through the process, so it would be easy to detect (learn) what could be happening.

After renaming the .ymal (e.g. fraunces.upstream.ymal) I would run:
`gftools packager mastering/googlefonts/fraunces.upstream.yaml ~/type-repos/google-font-repos/fonts -p --no-source`

This last flag is temporarily needed due to a gf backend issue.

**Comment by arrowtype** [comment]
Thanks, @vv-monsalve! I appreciate the help and tips here.

Unfortunately, I do think my main problem now is that I just don’t have push access to google/fonts. Hopefully @davelab6 can grant that soon.

```
ERROR: Permission to google/fonts.git denied to arrowtype.
fatal: Could not read from remote repository.

Please make sure you have the correct access rights
```

By the way, you said, `The first thing I notice is the .yaml doesn't have the family name, which from what I understood is needed as the tool would, by default, create/use a branch named out of the family name.` However, Marc’s latest PR has a file named just as `upstream.yaml`, so I think the name is just pulled from the `name` field within the YAML itself.

**Comment by graphicore** [comment]
In that yaml file linked above I spotted a line:

> `# Choose one of: {CATEGORIES}`

But that should be:

> ` # Choose one of: DISPLAY, SERIF, SANS_SERIF, HANDWRITING, MONOSPACE`

It's from an old version that has been fixed in 9298eca

**Comment by graphicore** [comment]
@vv-monsalve just mentions that the permission error could be SSH releated. @arrowtype are you using SSH in git to connect to github, e.g. when pushing?

**Comment by arrowtype** [comment]
It was a push access issue. Thanks!

**Comment by graphicore** [comment]
In that yaml file linked above I spotted a line:

> `# Choose one of: {CATEGORIES}`

But that should be:

> ` # Choose one of: DISPLAY, SERIF, SANS_SERIF, HANDWRITING, MONOSPACE`

It's from an old version that has been fixed in 9298eca
